src/ess/bifrost/workflow.py

```diff
@@ -87,6 +90,60 @@ def BifrostSimulationWorkflow(
     return workflow
 
 
+def BifrostWorkflow(
+    detector_names: list[NeXusDetectorName],
+) -> sciline.Pipeline:
+    """Data reduction workflow for BIFROST."""
+    workflow = TofWorkflow(
+        run_types=(SampleRun,),
+        monitor_types=(FrameMonitor0, FrameMonitor1, FrameMonitor2, FrameMonitor3),
+    )
+    # TODO change to use non-simulation providers
+    for provider in _SIMULATION_PROVIDERS:
+        workflow.insert(provider)
+    # TODO change to use non-simulation parameters
+    for key, val in simulation_default_parameters().items():
+        workflow[key] = val
+
+    workflow[BeamlineWithSpectrometerCoords[SampleRun]] = (
+        workflow[BeamlineWithSpectrometerCoords[SampleRun]]
+        .map(_make_detector_name_mapping(detector_names))
+        .reduce(func=merge_triplets)
+    )
+
+    workflow[NeXusData[NXdetector, SampleRun]] = (
+        workflow[NeXusData[NXdetector, SampleRun]]
+        .map(_make_detector_name_mapping(detector_names))
+        .reduce(func=concat_event_lists)
+    )
+
+    return workflow
+
+
+# TODO remove or move
+def concat_event_lists(
+    *data: sc.DataArray,
+) -> sc.DataArray:
+    """Concatenate binned event lists into a single data array in 'event_time_zero'.
+
+    Note that the output will likely have repeated values for 'event_time_zero'.
+    E.g., if input 1 has times `[0, 1, 2]` and input 2 has times `[0, 2, 3]`,
+    the output will have times `[0, 1, 2, 0, 2, 3]`.
```

Comment:
```suggestion
    the output will have times `[0, 1, 2, 0, 2, 3]`.
    Note that this sawtooth pattern will disappear again after grouping into pixels.
    Preserving it will likely actually lead to more efficient memory access patterns when grouping.
```
or something like that, since that insight we had in the discussion the other day was only obvious in hindsight.